lib/manager/components/DeploymentConfirmModal.js

```diff
@@ -0,0 +1,154 @@
+import React, {Component, PropTypes} from 'react'
+import {Alert as BootstrapAlert, Button, Glyphicon, Modal} from 'react-bootstrap'
+import {Map, TileLayer, Rectangle} from 'react-leaflet'
+
+import {getComponentMessages, getMessage} from '../../common/util/config'
+import {defaultTileURL} from '../../common/util/maps'
+import {versionHasExpired} from '../util'
+import polygon from 'turf-polygon'
+import area from '@turf/area'
+
+const BOUNDS_LIMIT = 10 // Limit for the decimal degrees span
+
+export default class DeploymentConfirmModal extends Component {
+  static propTypes = {
+    deployment: PropTypes.object,
+    deployToTargetClicked: PropTypes.func,
+    target: PropTypes.string
+  }
+
+  state = {}
+
+  messages = getComponentMessages('DeploymentConfirmModal')
+
+  _onClickDeploy = () => {
+    const {deployment, deployToTargetClicked, target} = this.props
+    deployToTargetClicked(deployment, target)
+    this.setState({isDeployed: true})
+  }
+
+  _onClose = () => this.props.onClose()
+
+  _getFeedNames = (versions) => versions.map(v => v.feedSource.name).join(', ')
```

Comment:
All functions that are not dependent on instance variables should be singleton functions. ie: define this as a function in the base of this file so a new function isn't created for each instance.